models/autoencoders/discreteVAE.py

```diff
@@ -84,21 +83,20 @@ def create_networks(self):
         logger.debug("Creating Network Structures")
         self.encoder=self._create_encoder()
         self.prior=self._create_prior()
-        self.sampler=self._create_sampler()
         self.decoder=self._create_decoder()
-        return
+        self.sampler = self._create_sampler(rbm=self.prior)
 
-    def _create_sampler(self):
+    def _create_sampler(self, rbm=None):
         logger.debug("Creating sampling routine")
         """
             Define the sampler to be used for sampling from the RBM.
 
         Returns:
-            Instance of GibbsSampler.
+            Instance of BaseSampler.
         """
-        assert self.prior is not None, "Prior (RBM) must be defined."
-        return GibbsSampler(RBM=self.prior, n_gibbs_sampling_steps=self._config.engine.n_gibbs_sampling_steps)
-    
+        assert rbm is not None, "Prior (RBM) must be defined."
+        return instantiate(self._config.sampler,RBM=self.prior)
```

Comment:
Here's where the magic happens: call the `instantiate` method with the config key `sampler`. This will read in all specified init arguments. Note that you can provide additional arguments like `RBM` during runtime too.